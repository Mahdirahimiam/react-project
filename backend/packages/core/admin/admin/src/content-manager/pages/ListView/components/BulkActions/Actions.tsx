import * as React from 'react';

import {
  Box,
  Button,
  Dialog,
  DialogBody,
  DialogFooter,
  DialogProps,
  Flex,
  ModalHeader,
  ModalLayout,
  Typography,
} from '@strapi/design-system';
import {
  useStrapiApp,
  useNotification,
  useTableContext,
  useFetchClient,
  useAPIErrorHandler,
  useTracking,
} from '@strapi/helper-plugin';
import { Check, ExclamationMarkCircle, Trash } from '@strapi/icons';
import { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
import { Common } from '@strapi/types';
import { AxiosError, AxiosResponse } from 'axios';
import { useIntl } from 'react-intl';
import { useQueryClient, useMutation } from 'react-query';
import { useParams } from 'react-router-dom';

import { DescriptionComponentRenderer } from '../../../../../components/DescriptionComponentRenderer';
import { useTypedSelector } from '../../../../../core/store/hooks';
import { getTranslation } from '../../../../utils/translations';
import { useAllowedActions } from '../../hooks/useAllowedActions';

import { PublishAction } from './PublishAction';

import type {
  BulkActionComponent,
  ContentManagerPlugin,
} from '../../../../../core/apis/content-manager';

interface BulkActionDescription {
  dialog?: DialogOptions | NotificationOptions | ModalOptions;
  disabled?: boolean;
  icon?: React.ReactNode;
  label: string;
  onClick?: (event: React.SyntheticEvent) => void;
  /**
   * @default 'default'
   */
  type?: 'icon' | 'default';
  /**
   * @default 'secondary'
   */
  variant?: 'default' | 'secondary' | 'tertiary' | 'danger-light' | 'success';
}

interface DialogOptions {
  type: 'dialog';
  title: string;
  content?: React.ReactNode;
  onConfirm?: () => void | Promise<void>;
  onCancel?: () => void | Promise<void>;
}

interface NotificationOptions {
  type: 'notification';
  title: string;
  link?: {
    label: string;
    url: string;
    target?: string;
  };
  content?: string;
  onClose?: () => void;
  status?: 'info' | 'warning' | 'softWarning' | 'success';
  timeout?: number;
}

interface ModalOptions {
  type: 'modal';
  title: string;
  content: React.ComponentType<{ onClose: () => void }>;
  onClose?: () => void;
}

/* -------------------------------------------------------------------------------------------------
 * BulkActionsRenderer
 * -----------------------------------------------------------------------------------------------*/

const BulkActionsRenderer = () => {
  const { plugins } = useStrapiApp();
  const { selectedEntries } = useTableContext();

  const { slug, collectionType } = useParams<{
    slug: Common.UID.ContentType;
    collectionType: string;
  }>();

  return (
    <Flex gap={2}>
      <DescriptionComponentRenderer
        props={{
          model: slug,
          collectionType,
          ids: selectedEntries,
        }}
        descriptions={(
          plugins['content-manager'].apis as ContentManagerPlugin['config']['apis']
        ).getBulkActions()}
      >
        {(actions) => actions.map((action) => <BulkActionAction key={action.id} {...action} />)}
      </DescriptionComponentRenderer>
    </Flex>
  );
};

/* -------------------------------------------------------------------------------------------------
 * BulkActionAction
 * -----------------------------------------------------------------------------------------------*/

interface Action extends BulkActionDescription, Pick<BulkActionComponent, 'actionType'> {
  id: string;
}

const BulkActionAction = (action: Action) => {
  const [dialogId, setDialogId] = React.useState<string | null>(null);
  const toggleNotification = useNotification();
  const { trackUsage } = useTracking();

  const handleClick = (action: Action) => (e: React.MouseEvent) => {
    const { onClick, dialog, id } = action;

    if (onClick) {
      onClick(e);
    }

    if (dialog) {
      switch (dialog.type) {
        case 'notification':
          toggleNotification({
            title: dialog.title,
            message: dialog.content,
            type: dialog.status,
            timeout: dialog.timeout,
            onClose: dialog.onClose,
          });
          break;
        case 'dialog':
        case 'modal': {
          if (action.actionType === 'delete') trackUsage('willBulkDeleteEntries');
          e.preventDefault();
          setDialogId(id);
        }
      }
    }
  };

  const handleClose = () => {
    setDialogId(null);
    if (action.dialog?.type === 'modal' && action.dialog?.onClose) {
      action.dialog.onClose();
    }
  };

  return (
    <>
      <Button
        disabled={action.disabled}
        startIcon={action.icon}
        variant={action.variant}
        onClick={handleClick(action)}
      >
        {action.label}
      </Button>
      {action.dialog?.type === 'dialog' ? (
        <BulkActionConfirmDialog
          {...action.dialog}
          variant={action.variant}
          isOpen={dialogId === action.id}
          onClose={handleClose}
        />
      ) : null}
      {action.dialog?.type === 'modal' ? (
        <BulkActionModal
          {...action.dialog}
          onModalClose={handleClose}
          isOpen={dialogId === action.id}
        />
      ) : null}
    </>
  );
};

/* -------------------------------------------------------------------------------------------------
 * BulkActionConfirmDialog
 * -----------------------------------------------------------------------------------------------*/

interface BulkActionConfirmDialogProps
  extends DialogOptions,
    Pick<DialogProps, 'onClose' | 'isOpen'>,
    Pick<Action, 'variant'> {
  confirmButton?: string;
}

const BulkActionConfirmDialog = ({
  onClose,
  onCancel,
  onConfirm,
  title,
  content,
  confirmButton,
  isOpen,
  variant = 'secondary',
}: BulkActionConfirmDialogProps) => {
  const { formatMessage } = useIntl();

  const handleClose = async () => {
    if (onCancel) {
      await onCancel();
    }

    onClose();
  };

  const handleConfirm = async () => {
    if (onConfirm) {
      await onConfirm();
    }

    onClose();
  };

  return (
    <Dialog isOpen={isOpen} title={title} onClose={handleClose}>
      <DialogBody icon={<ExclamationMarkCircle />}>{content}</DialogBody>
      <DialogFooter
        startAction={
          <Button onClick={handleClose} variant="tertiary">
            {formatMessage({
              id: 'app.components.Button.cancel',
              defaultMessage: 'Cancel',
            })}
          </Button>
        }
        endAction={
          <Button
            onClick={handleConfirm}
            variant={variant === 'danger-light' ? variant : 'secondary'}
            startIcon={variant === 'danger-light' ? <Trash /> : <Check />}
          >
            {confirmButton
              ? confirmButton
              : formatMessage({
                  id: 'app.components.Button.confirm',
                  defaultMessage: 'Confirm',
                })}
          </Button>
        }
      />
    </Dialog>
  );
};

/* -------------------------------------------------------------------------------------------------
 * BulkActionModal
 * -----------------------------------------------------------------------------------------------*/

interface BulkActionModalProps extends ModalOptions {
  onModalClose: () => void;
  isOpen?: boolean;
}

const BulkActionModal = ({
  isOpen,
  title,
  onClose,
  content: Content,
  onModalClose,
}: BulkActionModalProps) => {
  const id = React.useId();

  if (!isOpen) {
    return null;
  }

  const handleClose = () => {
    if (onClose) {
      onClose();
    }

    onModalClose();
  };

  return (
    <ModalLayout borderRadius="4px" overflow="hidden" onClose={handleClose} labelledBy={id}>
      <ModalHeader>
        <Typography fontWeight="bold" textColor="neutral800" as="h2" id={id}>
          {title}
        </Typography>
      </ModalHeader>
      <Content onClose={handleClose} />
    </ModalLayout>
  );
};

/* -------------------------------------------------------------------------------------------------
 * DefaultBulkActions
 * -----------------------------------------------------------------------------------------------*/

const DeleteAction: BulkActionComponent = ({ ids, model: slug }) => {
  const { formatMessage } = useIntl();
  const { post } = useFetchClient();
  const toggleNotification = useNotification();
  const { formatAPIError } = useAPIErrorHandler(getTranslation);
  const contentType = useTypedSelector((state) => state['content-manager_listView'].contentType);
  const { setSelectedEntries } = useTableContext();

  const hasI18nEnabled = Boolean(contentType?.pluginOptions?.i18n);
  const queryClient = useQueryClient();
  const { trackUsage } = useTracking();
  const hasDeletePermission = useAllowedActions(slug).canDelete;

  const handleConfirmDeleteAllData = async () => {
    try {
      await post<
        Contracts.CollectionTypes.BulkDelete.Response,
        AxiosResponse<Contracts.CollectionTypes.BulkDelete.Response>,
        Contracts.CollectionTypes.BulkDelete.Request['body']
      >(`/content-manager/collection-types/${slug}/actions/bulkDelete`, {
        ids,
      });

      queryClient.invalidateQueries(['content-manager', 'collection-types', slug]);
      setSelectedEntries([]);
      trackUsage('didBulkDeleteEntries');
    } catch (err) {
      if (err instanceof AxiosError) {
        toggleNotification({
          type: 'warning',
          message: formatAPIError(err),
        });
      }
    }
  };

  if (!hasDeletePermission) return null;

  return {
    actionType: 'delete',
    variant: 'danger-light',
    label: formatMessage({ id: 'global.delete', defaultMessage: 'Delete' }),
    dialog: {
      type: 'dialog',
      title: formatMessage({
        id: 'app.components.ConfirmDialog.title',
        defaultMessage: 'Confirmation',
      }),
      content: (
        <Flex direction="column" alignItems="stretch" gap={2}>
          <Typography id="confirm-description" textAlign="center">
            {formatMessage({
              id: 'popUpWarning.bodyMessage.contentType.delete.all',
              defaultMessage: 'Are you sure you want to delete these entries?',
            })}
          </Typography>
          {hasI18nEnabled && (
            <Box textAlign="center" padding={3}>
              <Typography textColor="danger500">
                {formatMessage(
                  {
                    id: getTranslation('Settings.list.actions.deleteAdditionalInfos'),
                    defaultMessage:
                      'This will delete the active locale versions <em>(from Internationalization)</em>',
                  },
                  {
                    em: Emphasis,
                  }
                )}
              </Typography>
            </Box>
          )}
        </Flex>
      ),
      onConfirm: handleConfirmDeleteAllData,
    },
  };
};

const UnpublishAction: BulkActionComponent = ({ ids, model: slug }) => {
  const { formatMessage } = useIntl();
  const { post } = useFetchClient();
  const toggleNotification = useNotification();
  const { formatAPIError } = useAPIErrorHandler(getTranslation);
  const { selectedEntries, setSelectedEntries } = useTableContext();
  const { data, contentType } = useTypedSelector((state) => state['content-manager_listView']);
  const selectedEntriesObjects = data.filter((entry) => selectedEntries.includes(entry.id));

  const hasI18nEnabled = Boolean(contentType?.pluginOptions?.i18n);
  const queryClient = useQueryClient();
  const hasPublishPermission = useAllowedActions(slug).canPublish;

  const bulkUnpublishMutation = useMutation<
    Contracts.CollectionTypes.BulkUnpublish.Response,
    AxiosError<Required<Pick<Contracts.CollectionTypes.BulkUnpublish.Response, 'error'>>>,
    Contracts.CollectionTypes.BulkUnpublish.Request['body']
  >(
    async (body) => {
      const { data } = await post<
        Contracts.CollectionTypes.BulkUnpublish.Response,
        AxiosResponse<Contracts.CollectionTypes.BulkUnpublish.Response>,
        Contracts.CollectionTypes.BulkUnpublish.Request['body']
      >(`/content-manager/collection-types/${slug}/actions/bulkUnpublish`, body);

      return data;
    },
    {
      onSuccess() {
        toggleNotification({
          type: 'success',
          message: {
            id: 'content-manager.success.record.unpublish',
            defaultMessage: 'Unpublished',
          },
        });

        queryClient.invalidateQueries(['content-manager', 'collection-types', slug]);
        setSelectedEntries([]);
      },
      onError(error) {
        toggleNotification({
          type: 'warning',
          message: formatAPIError(error),
        });
      },
    }
  );

  const handleConfirmUnpublishAllData = async () => {
    await bulkUnpublishMutation.mutateAsync({ ids });
  };

  const showUnpublishButton =
    hasPublishPermission && selectedEntriesObjects.some((entry) => entry.publishedAt);

  if (!showUnpublishButton) return null;

  return {
    actionType: 'unpublish',
    variant: 'tertiary',
    label: formatMessage({ id: 'app.utils.unpublish', defaultMessage: 'Unpublish' }),
    dialog: {
      type: 'dialog',
      title: formatMessage({
        id: 'app.components.ConfirmDialog.title',
        defaultMessage: 'Confirmation',
      }),
      content: (
        <Flex direction="column" alignItems="stretch" gap={2}>
          <Typography id="confirm-description" textAlign="center">
            {formatMessage({
              id: 'popUpWarning.bodyMessage.contentType.unpublish.all',
              defaultMessage: 'Are you sure you want to unpublish these entries?',
            })}
          </Typography>
          {hasI18nEnabled && (
            <Box textAlign="center" padding={3}>
              <Typography textColor="danger500">
                {formatMessage(
                  {
                    id: getTranslation('Settings.list.actions.unpublishAdditionalInfos'),
                    defaultMessage:
                      'This will unpublish the active locale versions <em>(from Internationalization)</em>',
                  },
                  {
                    em: Emphasis,
                  }
                )}
              </Typography>
            </Box>
          )}
        </Flex>
      ),
      confirmButton: formatMessage({
        id: 'app.utils.unpublish',
        defaultMessage: 'Unpublish',
      }),
      onConfirm: handleConfirmUnpublishAllData,
    },
  };
};

const Emphasis = (chunks: React.ReactNode) => (
  <Typography fontWeight="semiBold" textColor="danger500">
    {chunks}
  </Typography>
);

const DEFAULT_BULK_ACTIONS: BulkActionComponent[] = [PublishAction, UnpublishAction, DeleteAction];

export { DEFAULT_BULK_ACTIONS, BulkActionsRenderer, Emphasis };
export type { BulkActionDescription };
