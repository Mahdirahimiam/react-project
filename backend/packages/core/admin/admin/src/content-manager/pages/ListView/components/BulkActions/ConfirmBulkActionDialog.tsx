import * as React from 'react';

import {
  Button,
  Flex,
  Dialog,
  DialogBody,
  DialogFooter,
  Typography,
  DialogFooterProps,
} from '@strapi/design-system';
import {
  useTableContext,
  useFetchClient,
  useNotification,
  useAPIErrorHandler,
  useQueryParams,
} from '@strapi/helper-plugin';
import { Check, ExclamationMarkCircle } from '@strapi/icons';
import { useIntl } from 'react-intl';
import { useQuery } from 'react-query';

import { useTypedSelector } from '../../../../../core/store/hooks';
import { getTranslation } from '../../../../utils/translations';

import { Emphasis } from './Actions';

import type { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
import type { AxiosError } from 'axios';

interface ConfirmBulkActionDialogProps extends Pick<DialogFooterProps, 'endAction'> {
  onToggleDialog: () => void;
  isOpen?: boolean;
  dialogBody: React.ReactNode;
}

const ConfirmBulkActionDialog = ({
  onToggleDialog,
  isOpen = false,
  dialogBody,
  endAction,
}: ConfirmBulkActionDialogProps) => {
  const { formatMessage } = useIntl();

  return (
    <Dialog
      onClose={onToggleDialog}
      title={formatMessage({
        id: 'app.components.ConfirmDialog.title',
        defaultMessage: 'Confirmation',
      })}
      isOpen={isOpen}
    >
      <DialogBody icon={<ExclamationMarkCircle />}>
        <Flex direction="column" alignItems="stretch" gap={2}>
          {dialogBody}
        </Flex>
      </DialogBody>
      <DialogFooter
        startAction={
          <Button onClick={onToggleDialog} variant="tertiary">
            {formatMessage({
              id: 'app.components.Button.cancel',
              defaultMessage: 'Cancel',
            })}
          </Button>
        }
        endAction={endAction}
      />
    </Dialog>
  );
};

/* -------------------------------------------------------------------------------------------------
 * BoldChunk
 * -----------------------------------------------------------------------------------------------*/

const BoldChunk = (chunks: React.ReactNode) => <Typography fontWeight="bold">{chunks}</Typography>;

/* -------------------------------------------------------------------------------------------------
 * ConfirmDialogPublishAll
 * -----------------------------------------------------------------------------------------------*/

interface ConfirmDialogPublishAllProps
  extends Pick<ConfirmBulkActionDialogProps, 'isOpen' | 'onToggleDialog'> {
  isConfirmButtonLoading?: boolean;
  onConfirm: () => void;
}

const ConfirmDialogPublishAll = ({
  isOpen,
  onToggleDialog,
  isConfirmButtonLoading = false,
  onConfirm,
}: ConfirmDialogPublishAllProps) => {
  const { formatMessage } = useIntl();
  const { get } = useFetchClient();
  const { selectedEntries } = useTableContext();
  const toggleNotification = useNotification();
  const { formatAPIError } = useAPIErrorHandler(getTranslation);
  const contentType = useTypedSelector((state) => state['content-manager_listView'].contentType);
  const [{ query }] = useQueryParams<{
    plugins?: {
      i18n?: {
        locale?: string;
      };
    };
  }>();

  const slug = contentType?.uid ?? '';
  const hasI18nEnabled = Boolean(contentType?.pluginOptions?.i18n);

  const {
    data: countDraftRelations = 0,
    isLoading,
    isError,
  } = useQuery<
    Contracts.CollectionTypes.CountManyEntriesDraftRelations.Response['data'],
    AxiosError<
      Required<Pick<Contracts.CollectionTypes.CountManyEntriesDraftRelations.Response, 'error'>>
    >
  >(
    ['content-manager', 'draft-relations', slug, selectedEntries],
    async () => {
      const {
        data: { data },
      } = await get<Contracts.CollectionTypes.CountManyEntriesDraftRelations.Response>(
        `/content-manager/collection-types/${slug}/actions/countManyEntriesDraftRelations`,
        {
          params: {
            ids: selectedEntries,
            locale: query?.plugins?.i18n?.locale,
          },
        }
      );

      return data;
    },
    {
      // The API is called everytime you select/deselect an entry, this check avoids us sending a query with bad data
      enabled: selectedEntries.length > 0,
      onError(error) {
        toggleNotification({ type: 'warning', message: formatAPIError(error) });
      },
    }
  );

  if (isError) {
    return null;
  }

  return (
    <ConfirmBulkActionDialog
      isOpen={isOpen && !isLoading}
      onToggleDialog={onToggleDialog}
      dialogBody={
        <>
          <Typography id="confirm-description" textAlign="center">
            {countDraftRelations > 0 &&
              formatMessage(
                {
                  id: getTranslation(`popUpwarning.warning.bulk-has-draft-relations.message`),
                  defaultMessage:
                    '<b>{count} {count, plural, one { relation } other { relations } } out of {entities} { entities, plural, one { entry } other { entries } } {count, plural, one { is } other { are } }</b> not published yet and might lead to unexpected behavior. ',
                },
                {
                  b: BoldChunk,
                  count: countDraftRelations,
                  entities: selectedEntries.length,
                }
              )}
            {formatMessage({
              id: getTranslation('popUpWarning.bodyMessage.contentType.publish.all'),
              defaultMessage: 'Are you sure you want to publish these entries?',
            })}
          </Typography>
          {hasI18nEnabled && (
            <Typography textColor="danger500">
              {formatMessage(
                {
                  id: getTranslation('Settings.list.actions.publishAdditionalInfos'),
                  defaultMessage:
                    'This will publish the active locale versions <em>(from Internationalization)</em>',
                },
                {
                  em: Emphasis,
                }
              )}
            </Typography>
          )}
        </>
      }
      endAction={
        <Button
          onClick={onConfirm}
          variant="secondary"
          startIcon={<Check />}
          loading={isConfirmButtonLoading}
        >
          {formatMessage({
            id: 'app.utils.publish',
            defaultMessage: 'Publish',
          })}
        </Button>
      }
    />
  );
};

export { ConfirmDialogPublishAll, ConfirmBulkActionDialog };
export type { ConfirmDialogPublishAllProps, ConfirmBulkActionDialogProps };
