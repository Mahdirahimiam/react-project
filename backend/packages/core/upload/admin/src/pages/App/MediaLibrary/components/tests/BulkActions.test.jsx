import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';

import { BulkActions } from '../BulkActions';

jest.mock('../../../../../hooks/useBulkRemove');

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useNotification: jest.fn(),
  useQueryParams: jest.fn().mockReturnValue([{ query: {}, rawQuery: '' }, jest.fn()]),
}));

const setup = (
  props = {
    selected: [],
    onSuccess: jest.fn(),
  }
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <BulkActions {...props} />
          </IntlProvider>
        </MemoryRouter>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

describe('BulkActions', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('displays the proper assets and folders count', () => {
    const ASSET_COUNT = 2;
    const FOLDER_COUNT = 8;

    const { queryByText } = setup({
      onSuccess: jest.fn(),
      selected: [
        ...[...Array(ASSET_COUNT).keys()].map((index) => ({ id: index, type: 'asset' })),
        ...[...Array(FOLDER_COUNT).keys()].map((index) => ({ id: index, type: 'folder' })),
      ],
    });

    expect(
      queryByText(`${FOLDER_COUNT} folders - ${ASSET_COUNT} assets selected`)
    ).toBeInTheDocument();
  });
});
