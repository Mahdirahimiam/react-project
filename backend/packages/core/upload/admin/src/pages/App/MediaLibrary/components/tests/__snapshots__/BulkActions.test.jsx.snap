// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkActions renders 1`] = `
.c9 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c3 {
  font-size: 1rem;
  line-height: 1.5;
  color: #666687;
}

.c7 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c0 {
  padding-bottom: 20px;
}

.c4 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c5 {
  position: relative;
  outline: none;
}

.c5 > svg {
  height: 12px;
  width: 12px;
}

.c5 > svg > g,
.c5 > svg path {
  fill: #ffffff;
}

.c5[aria-disabled='true'] {
  pointer-events: none;
}

.c5:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c5:focus-visible {
  outline: none;
}

.c5:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c6 {
  height: 2rem;
  border: 1px solid #f5c0b8;
  background: #fcecea;
}

.c6 svg {
  height: 0.75rem;
  width: auto;
}

.c6[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c6[aria-disabled='true'] .c2 {
  color: #666687;
}

.c6[aria-disabled='true'] svg > g,
.c6[aria-disabled='true'] svg path {
  fill: #666687;
}

.c6[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c6[aria-disabled='true']:active .c2 {
  color: #666687;
}

.c6[aria-disabled='true']:active svg > g,
.c6[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c6:hover {
  background-color: #ffffff;
}

.c6:active {
  background-color: #ffffff;
  border: 1px solid #d02b20;
}

.c6:active .c2 {
  color: #d02b20;
}

.c6:active svg > g,
.c6:active svg path {
  fill: #d02b20;
}

.c6 .c2 {
  color: #b72b1a;
}

.c6 svg > g,
.c6 svg path {
  fill: #b72b1a;
}

.c8 {
  height: 2rem;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c8 svg {
  height: 0.75rem;
  width: auto;
}

.c8[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c8[aria-disabled='true'] .c2 {
  color: #666687;
}

.c8[aria-disabled='true'] svg > g,
.c8[aria-disabled='true'] svg path {
  fill: #666687;
}

.c8[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c8[aria-disabled='true']:active .c2 {
  color: #666687;
}

.c8[aria-disabled='true']:active svg > g,
.c8[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c8:hover {
  background-color: #ffffff;
}

.c8:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c8:active .c2 {
  color: #4945ff;
}

.c8:active svg > g,
.c8:active svg path {
  fill: #4945ff;
}

.c8 .c2 {
  color: #271fe0;
}

.c8 svg > g,
.c8 svg path {
  fill: #271fe0;
}

<div>
  <div
    class="c0 c1"
  >
    <span
      class="c2 c3"
    >
      0 folders - 0 assets selected
    </span>
    <button
      aria-disabled="false"
      class="c4 c1 c5 c6"
      type="button"
    >
      <div
        aria-hidden="true"
        class=""
      >
        <svg
          fill="none"
          height="1rem"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.236 6.149a.2.2 0 0 0-.197.233L6 24h12l2.96-17.618a.2.2 0 0 0-.196-.233H3.236ZM21.8 1.983c.11 0 .2.09.2.2v1.584a.2.2 0 0 1-.2.2H2.2a.2.2 0 0 1-.2-.2V2.183c0-.11.09-.2.2-.2h5.511c.9 0 1.631-1.09 1.631-1.983h5.316c0 .894.73 1.983 1.631 1.983H21.8Z"
            fill="#32324D"
          />
        </svg>
      </div>
      <span
        class="c2 c7"
      >
        Delete
      </span>
    </button>
    <button
      aria-disabled="false"
      class="c4 c1 c5 c8"
      type="button"
    >
      <div
        aria-hidden="true"
        class=""
      >
        <svg
          fill="none"
          height="1rem"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.414 5H21a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h7.414l2 2Z"
            fill="#212134"
          />
        </svg>
      </div>
      <span
        class="c2 c7"
      >
        Move
      </span>
    </button>
  </div>
  <div
    class="c9"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
